[ZLIB] Update to 1.2.12, CORE-18339

pmeerw opened the pull request:
Update zlib files to 1.2.12 from upstream, commit 21767c654d31d2dccdde4330529775c6c5fd5389

JIRA issue: [CORE-18339](https://jira.reactos.org/browse/CORE-18339)

Fixes widely publicized deflate bug in zlib. Faster CRC-32.

SergeGautherie commented:
Add 'CORE-18339' to commit comments.

> [ZLIB] Update to upstream v1.2.12, commit 21767c654d31d2dccdde4330529…

This commit is the release one:
it may not be needed to be that explicit,
at least, not needed on the commit title line.
https://github.com/madler/zlib/releases/tag/v1.2.12

Please, also update
https://github.com/reactos/reactos/blob/d4cab249ef1318ac9e9a838841d53f9f15bc3f58/media/doc/3rd%20Party%20Files.txt#L254

> ZLIB] Restore 'uncrypt' functionality in minizip (using #ifdef REACTOS)

Typo: missing open bracket.

pmeerw commented:
@SergeGautherie addressed your code comments

> Does ReactOS actually need/want this uncrypt feature? If so, could you add a comment/example about that on commit comment? Thanks.

it appears to be used in dll/shellext/zipfldr/CZipExtract.cpp:                    err = unzOpenCurrentFilePassword(uf, Password);

HBelusca commented on sdk/lib/3rdparty/zlib/contrib/minizip/iowin32.c:
```diff
@@ -95,7 +95,6 @@ static voidpf win32_build_iowin(HANDLE hFile)
 
 voidpf ZCALLBACK win32_open64_file_func (voidpf opaque,const void* filename,int mode)
 {
-    //const char* mode_fopen = NULL;
```
Please revert.

pmeerw replied:
right, will comment out (as before) instead of removing line

HBelusca commented on sdk/lib/3rdparty/zlib/contrib/minizip/iowin32.c:
```diff
@@ -124,7 +123,6 @@ voidpf ZCALLBACK win32_open64_file_func (voidpf opaque,const void* filename,int
 
 voidpf ZCALLBACK win32_open64_file_funcA (voidpf opaque,const void* filename,int mode)
 {
-    //const char* mode_fopen = NULL;
```
Please revert.

HBelusca commented on sdk/lib/3rdparty/zlib/contrib/minizip/iowin32.c:
```diff
@@ -148,7 +146,6 @@ voidpf ZCALLBACK win32_open64_file_funcA (voidpf opaque,const void* filename,int
 
 voidpf ZCALLBACK win32_open64_file_funcW (voidpf opaque,const void* filename,int mode)
 {
-    //const char* mode_fopen = NULL;
```
Please revert.

HBelusca commented on sdk/lib/3rdparty/zlib/contrib/minizip/iowin32.c:
```diff
@@ -168,7 +165,6 @@ voidpf ZCALLBACK win32_open64_file_funcW (voidpf opaque,const void* filename,int
 
 voidpf ZCALLBACK win32_open_file_func (voidpf opaque,const char* filename,int mode)
 {
-    //const char* mode_fopen = NULL;
```
Please revert.

ThFabba reviewed (changes requested):
You're missing an update to `sdk/include/reactos/libs/zlib/zlib.h`

pmeerw commented on sdk/lib/3rdparty/zlib/contrib/minizip/zip.c:
```diff
@@ -523,7 +523,7 @@ local ZPOS64_T zip64local_SearchCentralDir(const zlib_filefunc64_32_def* pzlib_f
       if (((*(buf+i))==0x50) && ((*(buf+i+1))==0x4b) &&
```
the braces were explicitly added in Reactos in commit 1ac9e484657f12b548f8424b938ad6a5275ffc29 to apparently work around a warning in GCC 8, I suggest to keep

pmeerw replied:
zlib.h, uups

ThFabba replied:
The warning was fixed upstream by fixing the indent. So now https://github.com/reactos/reactos/commit/1ac9e484657f12b548f8424b938ad6a5275ffc29 is no longer needed.
But it doesn't really matter -- the braces will probably disappear on some future sync anyway.
